### PR TITLE
Export dataset in CVAT format misses frames in tasks with non-default…

### PR DESCRIPTION
**Pull request by bsekachev**
… frame step

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **Bug Fixes**
	- Resolved an issue where dataset exports in CVAT format were missing frames when using non-default frame steps.

- **Improvements**
	- Enhanced the efficiency of dataset export operations by optimizing how updated timestamps are retrieved.
	- Improved error handling for export cache management with more specific exception handling.

- **Changes**
	- Modified frame iteration logic to allow for broader frame processing without skipping based on frame steps.

### CODE REVIEW

File: cvat/apps/dataset_manager/views.py [hunk 1]
```diff
@@ -122,8 +122,8 @@ def export(dst_format, project_id=None, task_id=None, job_id=None, server_url=No
         instance_update_time = timezone.localtime(db_instance.updated_date)
         if isinstance(db_instance, Project):
             tasks_update = list(map(
-                lambda db_task: timezone.localtime(db_task.updated_date),
-                db_instance.tasks.all()
+                lambda updated_date: timezone.localtime(updated_date),
+                db_instance.tasks.values_list('updated_date', flat=True)
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/views.py:126]: It's even possible to request max directly from the DB. In either case, there will be new DB requests, as it's not `.all()`, which can be cached.

File: changelog.d/20241107_165701_sekachev.bs_fixed_export.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+### Fixed
+
+- Export dataset in CVAT format misses frames in tasks with non-default frame step
+  (<https://github.com/cvat-ai/cvat/pull/XXXX>)
```
Comment (zhiltsov-max) [inline changelog.d/20241107_165701_sekachev.bs_fixed_export.md:4]:
```suggestion
  (<https://github.com/cvat-ai/cvat/pull/8662>)
```

File: cvat/apps/engine/frame_provider.py [hunk 3]
```diff
@@ -369,7 +369,7 @@ def iterate_frames(
         quality: FrameQuality = FrameQuality.ORIGINAL,
         out_type: FrameOutputType = FrameOutputType.BUFFER,
     ) -> Iterator[DataWithMeta[AnyFrame]]:
-        frame_range = itertools.count(start_frame, self._db_task.data.get_frame_step())
+        frame_range = itertools.count(start_frame)
```
Comment (zhiltsov-max) [inline cvat/apps/engine/frame_provider.py:372]: Probably, `db_segment_frame_set` below has to be converted to relative ids. There is a class method for this

Comment (zhiltsov-max) [reply cvat/apps/engine/frame_provider.py:372]: Ok, please call `dev/format_python_code.sh`

**Comment by bsekachev** [comment]
Yes, it would be. Hovewer I do not have enough time. Let's keep the card on the agile board.

File: cvat/apps/engine/frame_provider.py (hunk 3 above)
Comment (bsekachev) [reply cvat/apps/engine/frame_provider.py:372]: Why not to add it to git pre-commit hook?

Comment (zhiltsov-max) [reply cvat/apps/engine/frame_provider.py:372]: There is a problem with determining the right interpreter in a cross-platform manner.

**Pull request merged**